--- lib/fribidi-common.h.orig	2012-10-16 12:59:40.666004678 -0600
+++ lib/fribidi-common.h	2012-10-16 13:01:29.421998633 -0600
@@ -52,13 +52,7 @@
 #endif /* !FRIBIDI_NAMESPACE */
 
 /* FRIBIDI_ENTRY is a macro used to declare library entry points. */
-#ifndef FRIBIDI_ENTRY
-# if (defined(WIN32)) || (defined(_WIN32_WCE))
-#  define FRIBIDI_ENTRY __declspec(dllimport)
-# else /* !WIN32 */
 #  define FRIBIDI_ENTRY		/* empty */
-# endif	/* !WIN32 */
-#endif /* !FRIBIDI_ENTRY */
 
 #if FRIBIDI_USE_GLIB+0
 # ifndef __FRIBIDI_DOC
