--- src/info/mkvinfo-org.cpp	2014-07-13 14:59:54.767079300 +0200
+++ src/info/mkvinfo.cpp	2014-07-13 15:00:00.677417300 +0200
@@ -395,30 +395,6 @@
   return false;
 }
 
-#if defined(COMP_MSC) || defined(COMP_MINGW)
-struct tm *
-gmtime_r(const time_t *timep,
-         struct tm *result) {
-  struct tm *aresult;
-
-  aresult = gmtime(timep);
-  memcpy(result, aresult, sizeof(struct tm));
-
-  return result;
-}
-
-char *
-asctime_r(const struct tm *tm,
-          char *buf) {
-  char *abuf;
-
-  abuf = asctime(tm);
-  strcpy(buf, abuf);
-
-  return abuf;
-}
-#endif
-
 void
 read_master(EbmlMaster *m,
             EbmlStream *es,
