From 79b536b9d8503a08a339b450df21f96aeb33d41a Mon Sep 17 00:00:00 2001
From: Ricardo Constantino <wiiaboo@gmail.com>
Date: Wed, 16 Dec 2015 18:17:38 +0000
Subject: [PATCH] waf: Use pkgconfig with ANGLE

---
 wscript | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/wscript b/wscript
index a5e6914..c826232 100644
--- a/wscript
+++ b/wscript
@@ -675,9 +675,7 @@ video_output_features = [
         'desc': 'OpenGL Win32 ANGLE Backend',
         'deps_any': [ 'os-win32', 'os-cygwin' ],
         'groups': [ 'gl' ],
-        'func': check_statement(['EGL/egl.h'],
-                                'eglCreateWindowSurface(0, 0, 0, 0)',
-                                lib='EGL')
+        'func': check_pkg_config('libEGL'),
     } , {
         'name': '--vdpau',
         'desc': 'VDPAU acceleration',
-- 
2.6.3

