From e821c0839c6c764896b0091c289527e485829b81 Mon Sep 17 00:00:00 2001
From: Ricardo Constantino <wiiaboo@gmail.com>
Date: Wed, 11 May 2016 16:15:12 +0100
Subject: [PATCH] waf: Use pkgconfig with ANGLE

---
 wscript | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/wscript b/wscript
index 6ff38d6..341479b 100644
--- a/wscript
+++ b/wscript
@@ -699,8 +699,7 @@ video_output_features = [
         'desc': 'OpenGL Win32 ANGLE Backend',
         'deps_any': [ 'os-win32', 'os-cygwin' ],
         'groups': [ 'gl' ],
-        'func': check_statement(['EGL/egl.h', 'EGL/eglext.h'],
-                                'int x = EGL_D3D_TEXTURE_2D_SHARE_HANDLE_ANGLE')
+        'func': check_pkg_config('libEGL'),
     } , {
         'name': '--vdpau',
         'desc': 'VDPAU acceleration',
-- 
2.8.1

