diff -ru SDL2-2.0.5-orig/src/core/windows/SDL_xinput.h SDL2-2.0.5/src/core/windows/SDL_xinput.h
--- SDL2-2.0.5-orig/src/core/windows/SDL_xinput.h	2016-10-20 05:56:26.000000000 +0200
+++ SDL2-2.0.5/src/core/windows/SDL_xinput.h	2017-02-14 06:51:51.651147735 +0100
@@ -102,24 +102,6 @@
 /* typedef's for XInput structs we use */
 typedef struct
 {
-    WORD wButtons;
-    BYTE bLeftTrigger;
-    BYTE bRightTrigger;
-    SHORT sThumbLX;
-    SHORT sThumbLY;
-    SHORT sThumbRX;
-    SHORT sThumbRY;
-    DWORD dwPaddingReserved;
-} XINPUT_GAMEPAD_EX;
-
-typedef struct
-{
-    DWORD dwPacketNumber;
-    XINPUT_GAMEPAD_EX Gamepad;
-} XINPUT_STATE_EX;
-
-typedef struct
-{
     BYTE BatteryType;
     BYTE BatteryLevel;
 } XINPUT_BATTERY_INFORMATION_EX;
