From 8e1d3b93ecd2294d900a141ae983d931c7e02eee Mon Sep 17 00:00:00 2001
From: shinchiro <shinchiro@nowhere.com>
Date: Wed, 23 Dec 2015 13:57:49 +0800
Subject: [PATCH 2/3] Simplify secure API functions workaround for MinGW

---
 src/libANGLE/renderer/d3d/RendererD3D.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/libANGLE/renderer/d3d/RendererD3D.cpp b/src/libANGLE/renderer/d3d/RendererD3D.cpp
index 163db09..3819381 100644
--- a/src/libANGLE/renderer/d3d/RendererD3D.cpp
+++ b/src/libANGLE/renderer/d3d/RendererD3D.cpp
@@ -6,6 +6,10 @@
 
 // RendererD3D.cpp: Implementation of the base D3D Renderer.
 
+#ifdef __MINGW32__
+#define MINGW_HAS_SECURE_API 1
+#endif
+
 #include "libANGLE/renderer/d3d/RendererD3D.h"
 
 #include "common/debug.h"
-- 
2.7.1

