From eb6416cf4b8c3e9f1a405b324160890a6ff71116 Mon Sep 17 00:00:00 2001
From: RiCON <RiCON@nowhere>
Date: Sun, 2 Aug 2015 00:51:12 +0100
Subject: [PATCH 1/2] Fix header conflicts with MinGW

---
 jmorecfg.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/jmorecfg.h b/jmorecfg.h
index 1d96786..7d908df 100644
--- a/jmorecfg.h
+++ b/jmorecfg.h
@@ -26,6 +26,12 @@
 
 #define MAX_COMPONENTS  10      /* maximum number of image components */
 
+/* prevents conflicts */
+#if defined(__MINGW32__)
+#include <shlwapi.h> /* typedefs INT16 and INT32 */
+
+#define HAVE_BOOLEAN
+#endif
 
 /*
  * Basic data types.
-- 
2.6.3

