From 50b2e0802f1036cbba3099de93307f4cb9dd846c Mon Sep 17 00:00:00 2001
From: "Ricardo Constantino (:RiCON)" <wiiaboo@gmail.com>
Date: Wed, 16 Mar 2016 14:34:03 +0000
Subject: [PATCH 4/4] CMake: Allow not building executable

---
 CMakeLists.txt     | 2 ++
 src/CMakeLists.txt | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 325f763..b53420d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -69,6 +69,8 @@ if (BUILD_SHARED_LIBS)
     option(BUILD_STATIC "Build static library"
            ON)
 endif (BUILD_SHARED_LIBS)
+option(BUILD_BINARY "Build executable"
+       ON)
 
 configure_file(
 	uchardet.pc.in
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 32e62c2..e50ae50 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -132,4 +132,6 @@ install(
 
 include(symbols.cmake)
 
+if (BUILD_BINARY)
 add_subdirectory(tools)
+endif (BUILD_BINARY)
-- 
2.7.3

