From 36665da832cca6621b86f05fb0ff64551c114051 Mon Sep 17 00:00:00 2001
From: "Ricardo Constantino (:RiCON)" <wiiaboo@gmail.com>
Date: Wed, 16 Mar 2016 14:17:25 +0000
Subject: [PATCH 1/4] CMake: allow installing binary to non-default dir

---
 CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4f279e1..223b53f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -37,6 +37,10 @@ set (DIR_SHARE ${DIR_PREFIX}/share)
 set (DIR_BIN ${DIR_PREFIX}/bin)
 set (DIR_ETC ${DIR_PREFIX}/etc)
 
+if (DEFINED CMAKE_INSTALL_BINDIR)
+	set (DIR_BIN ${CMAKE_INSTALL_BINDIR})
+endif (DEFINED CMAKE_INSTALL_BINDIR)
+
 if (DEFINED CMAKE_INSTALL_LIBDIR)
 	set (DIR_LIBRARY ${CMAKE_INSTALL_LIBDIR})
 	set (DIR_LIBRARY_STATIC ${CMAKE_INSTALL_LIBDIR})
-- 
2.7.3

