From f53cb8cdddf96de2540995fe01df482fc53a8b45 Mon Sep 17 00:00:00 2001
From: "Ricardo Constantino (:RiCON)" <wiiaboo@gmail.com>
Date: Wed, 16 Mar 2016 14:17:47 +0000
Subject: [PATCH 2/4] CMake: fix linking with Ninja

---
 src/symbols.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/symbols.cmake b/src/symbols.cmake
index 9453baf..bd83f1d 100644
--- a/src/symbols.cmake
+++ b/src/symbols.cmake
@@ -25,7 +25,7 @@ elseif (CMAKE_CXX_COMPILER_ID STREQUAL GNU)
 	set(_version_script "${CMAKE_CURRENT_BINARY_DIR}/version.script")
 	file(WRITE ${_version_script} "${_symbols}\n")
 
-	set(LINK_FLAGS "${LINK_FLAGS} -Wl,--version-script,'${_version_script}'")
+	set(LINK_FLAGS "${LINK_FLAGS} -Wl,--version-script,${_version_script}")
 	
 endif (APPLE)
 
-- 
2.7.3

