From 5eedeae4a013a6ffd92075be0c9edc90e8a8197f Mon Sep 17 00:00:00 2001
From: RiCON <wiiaboo@gmail.com>
Date: Sun, 6 Sep 2015 01:45:49 +0100
Subject: [PATCH] Fix compilation with ffmpeg git

---
 x264.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x264.c b/x264.c
index 4dd2a26..6609fef 100644
--- a/x264.c
+++ b/x264.c
@@ -442,7 +442,7 @@ static void print_csp_names( int longhelp )
     printf( "                              - valid csps for `lavf' demuxer:\n" );
     printf( INDENT );
     size_t line_len = strlen( INDENT );
-    for( enum PixelFormat i = AV_PIX_FMT_NONE+1; i < AV_PIX_FMT_NB; i++ )
+    for( enum AVPixelFormat i = AV_PIX_FMT_NONE+1; i < AV_PIX_FMT_NB; i++ )
     {
         const char *pfname = av_get_pix_fmt_name( i );
         if( pfname )
-- 
2.5.1

